### PR TITLE
Working IOI3 tile split.

### PR DESCRIPTION
**Pull request by litghost**
This is in support of the ROI breakout work.

### CODE REVIEW

File: utils/tile_splitter/grid.py [hunk 1]
```diff
@@ -397,7 +405,7 @@ def row(self, y):
         for _ in range(y):
             right_of_row = right_of_row.neighboors[SOUTH]
 
-    def split_tile(self, tile, tile_type_pkeys, split_direction):
+    def split_tile(self, tile, tile_type_pkeys, split_direction, split_map):
```
Comment (mkurc-ant) [inline utils/tile_splitter/grid.py]: I'd add split_map to the docstring and explain what it is.

File: utils/tile_splitter/grid.py [hunk 2]
```diff
@@ -475,36 +500,40 @@ def insert_empty_column(self, top_of_column, insert_in_direction):
...
+            split_map,
     ):
-        """ Split column of tiles.
+        """ Split row/column of tiles.
 
-        Splits specified tile types into new columns by first inserting any
-        required empty columns in the split direction, and then performing
+        Splits specified tile types into new row/column by first inserting any
+        required empty row/column in the split direction, and then performing
         the split.
```
Comment (mkurc-ant) [inline utils/tile_splitter/grid.py]: No docstring for split_map

File: xc7/utils/prjxray_form_channels.py [hunk 3]
```diff
@@ -1429,37 +1432,126 @@ def traverse_pip(conn, wire_in_tile_pkey):
...
-    write_cur.execute('INSERT INTO tile_type(name) VALUES ("SLICEL");')
-    slicel_tile_type_pkey = write_cur.lastrowid
-
-    write_cur.execute('INSERT INTO tile_type(name) VALUES ("SLICEM");')
-    slicem_tile_type_pkey = write_cur.lastrowid
+    # Insert synthetic tile types for CLB sites and iopad sites.
+    for name in [
+            'SLICEL',
+            'SLICEM',
+            'LIOPAD_M',
```
Comment (mkurc-ant) [inline xc7/utils/prjxray_form_channels.py]: Do we need to have different IOPAD tiles for L and R ?

Comment (litghost) [reply xc7/utils/prjxray_form_channels.py]: We don't need logical placement types for L and R, but the routing interface for L and R are different.

File: utils/tile_splitter/grid.py (hunk 1 above)
Comment (litghost) [reply utils/tile_splitter/grid.py]: Agreed

Comment (litghost) [reply utils/tile_splitter/grid.py]: Done

File: utils/tile_splitter/grid.py (hunk 2 above)
Comment (litghost) [reply utils/tile_splitter/grid.py]: Done

**Pull request merged**